packages/space-opera/src/components/model_viewer_snippet/reducer.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import {ModelViewerConfig} from '@google/model-viewer-editing-adapter/lib/main.js';
+
+import {registerStateMutator} from '../../space_opera_base.js';
+import {State} from '../../space_opera_base.js';
+import {INITIAL_CAMERA} from '../camera_settings/camera_state.js';
+
+// TODO: BUGGY! This is mutating the state!
```

Comment:
Was this here before or did you add it? Also, isn't a state mutator supposed to mutate the state?

Reply:
I added this. It is supposed to mutate the state, but it is supposed to mutate a copy of the original state. So a good number of the reducers are in fact "broken" by redux standards.